packages/js/src/utils/typescript/run-type-check.ts

```diff
@@ -103,9 +106,9 @@ export async function runTypeCheck(
 
   const result = program.emit();
 
-  const allDiagnostics = ts
-    .getPreEmitDiagnostics(program as Program)
-    .concat(result.diagnostics);
+  const allDiagnostics = options.ignoreDiagnostics
```

Comment:
Makes sense for correctness. We should look at it again in the future to see if want to allow builds without typechecks, maybe a separate `typedefs` task for swc and esbuild so users that do not need `d.ts` files don't need to pay the penalty.